Snake/snake.py

```diff
@@ -1,5 +1,48 @@
-import pygame
 import random
+from dataclasses import dataclass, field
+from typing import Callable
+
+import pygame
+
+
+# immutable Dataclass as a representation of postion and direction information
+@dataclass(frozen=True)
+class Position:
+    x: int
+    y: int
+
+    def __add__(
+        self,
+        o: "Position",
+    ) -> "Position":
+        return Position(self.x + o.x, self.y + o.y)
+
+    def apply_transform(self, f: Callable[[int], int] = lambda x: x):
+        return Position(f(self.x), f(self.y))
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, Position):
+            return NotImplemented
+        return (self.x - o.x, self.y - o.y) == (0, 0)
```

Comment:
Question for myself: The functions `__add__` and `__eq__`  arenot used in the code. Were the functions implemented because of a coding convention or for reusability reasons?

Reply:
@JulesOxe The __f__ methonds in python are special functions that can be used like this 
__eq__ is called when using the == operator on two Positions 
__add__ is used to add two Positions using the + operator

Reply:
All right, thanks. I'll remember that for the future.